Darwin/StitchCore/StitchCoreTests/Core/StitchAppClientIntegrationAuthTests.swift

```diff
@@ -171,22 +215,97 @@ class StitchAppClientIntegrationAuthTests: StitchIntegrationTestCase {
             XCTAssertNotEqual(anonUserID, nextUserID)
             emailUserID = nextUserID
 
-            self.verifyBasicAuthStorageInfo(loggedIn: true, expectedProviderType: StitchProviderType.userPassword)
+            self.verifyBasicAuthInfo(
+                loggedInList: [false, true, true],
+                loggedIn: true,
+                expectedProviderType: StitchProviderType.userPassword,
+                expectedUserIndex: 2)
+
             exp3.fulfill()
         }
         wait(for: [exp3], timeout: defaultTimeoutSeconds)
 
-        let exp4 = expectation(description: "logged in as second email/password user")
-        self.registerAndLogin(email: "test2@10gen.com", password: "hunter2") { user in
-            let nextUserID = user.id
-            XCTAssertNotEqual(emailUserID, nextUserID)
+        let exp4 = expectation(description: "switching to previous user should work properly")
```

Comment:
[note] definitely no need to change this since this is test code, but in the future you can use `CallbackJoiner` to make tests significantly less verbose. Jason added it recently. See `SyncIntTests.swift` and `RemoteMongoClientIntTests.swift` for usage examples